relayer/chains/icon/query.go

```diff
@@ -98,59 +98,59 @@ func (p *Provider) GenerateMessages(ctx context.Context, key *providerTypes.Mess
 			}
 			switch el.Indexed[0] {
 			case EmitMessage:
-				if el.Addr == types.Address(p.cfg.Contracts[providerTypes.ConnectionContract]) &&
```

Comment:
(len(el.Indexed) != 3 && len(el.Data) != 1)

Shouldn't this also be "or"?
Since we do not want to process if the event is malformed in any way.
Will most likely never happened but still good to be consistent

Reply:
Current change is just  unnecessarily stricter, yeah early exit is good.